File: frontend/lib/models/supplier_complaint_status_update_dto.dart
class SupplierComplaintStatusUpdateDto {
  final String status;
  final bool? resolved;
  final String? resolutionNotes;

  SupplierComplaintStatusUpdateDto({
    required this.status,
    this.resolved,
    this.resolutionNotes,
  });

  Map<String, dynamic> toJson() => {
        'status': status,
        if (resolved != null) 'resolved': resolved,
        if (resolutionNotes != null) 'resolutionNotes': resolutionNotes,
      };
}

File: backend/CoreService/src/main/java/ru/bowling/bowlingapp/DTO/SupplierComplaintStatusUpdateDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.bowling.bowlingapp.Entity.enums.SupplierComplaintStatus;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SupplierComplaintStatusUpdateDTO {

    @NotNull
    private SupplierComplaintStatus status;

    private Boolean resolved;

    private String resolutionNotes;
}

